src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,176 @@
+    isLenient = true
+}
+
+const val DERIBIT_TEST_WS_URL = "wss://test.deribit.com/ws/api/v2"
+const val INSTRUMENT_NAME_DEFAULT = "BTC-PERPETUAL"
+const val TICKER_SUBSCRIPTION_INTERVAL = "100ms"
+const val SLIDING_WINDOW_SECONDS = 10L
+const val TREND_OUTPUT_INTERVAL_MS = 1000L
+const val MIN_POINTS_FOR_TREND = 5
+
+val requestId = AtomicInteger(1)
+val priceDataWindow = ArrayDeque<PricePoint>()
+
+class TrendData(
+    val slope: Double,
+    val intercept: Double,
+    val currentPrice: Double,
+)
+
+fun main() = runBlocking {
+    println("Enter instrument name (e.g., BTC-PERPETUAL, ETH-PERPETUAL) or press Enter for default ($INSTRUMENT_NAME_DEFAULT):")
+    val instrumentName = readlnOrNull()?.takeIf { it.isNotBlank() } ?: INSTRUMENT_NAME_DEFAULT
+    println("Using instrument: $instrumentName")
+
+    val client = HttpClient(CIO) {
+        install(WebSockets)
+        install(ContentNegotiation) {
+            json(json)
+        }
+    }
+
+    val trendJob = launch {
+        var lastOutputTime = 0L
+        while (isActive) {
+            val currentTime = System.currentTimeMillis()
+            if (currentTime - lastOutputTime >= TREND_OUTPUT_INTERVAL_MS) {
+                val data = synchronized(priceDataWindow) {
+                    if (priceDataWindow.size >= MIN_POINTS_FOR_TREND) {
+                        val currentPoints = priceDataWindow.toList()
+                        val (s, i) = calculateLinearTrend(currentPoints)
+                        TrendData(s, i, currentPoints.last().price)
+                    } else {
+                        null
+                    }
+                }
+
+                if (data != null) {
+                    val trendDirection = when {
+                        data.slope > 0.00001 -> "UP"
+                        data.slope < -0.00001 -> "DOWN"
+                        else -> "FLAT"
+                    }
+                    val firstTimestamp = synchronized(priceDataWindow) { priceDataWindow.first().timestamp }
+                    val relativeTimeNow =
+                        (System.currentTimeMillis() - firstTimestamp).toDouble() / 1000.0 // in seconds
+                    val extrapolatedPrice = data.slope * (relativeTimeNow + 1.0) + data.intercept
+
+                    println(
+                        "Inst: $instrumentName | Window: ${SLIDING_WINDOW_SECONDS}s (${priceDataWindow.size} pts) | " +
+                                "Price: ${"%.2f".format(data.currentPrice)} | Trend: $trendDirection (${
+                                    "%.4f".format(data.slope)
+                                } price/sec) | " +
+                                "Extrapolated next (1s): ${"%.2f".format(extrapolatedPrice)}"
+                    )
+                    lastOutputTime = currentTime
+                } else {
+                    println(
+                        "Instrument: $instrumentName | Window: ${SLIDING_WINDOW_SECONDS}s (${priceDataWindow.size} pts) | (collecting data...)"
```

Comment:
Для однообразия сообщений:
```suggestion
                        "Inst: $instrumentName | Window: ${SLIDING_WINDOW_SECONDS}s (${priceDataWindow.size} pts) | (collecting data...)"
```